toxcore/group_chats.c

```diff
@@ -1495,9 +1503,16 @@ int group_packet_wrap(
         return -1;
     }
 
-    if (length > MAX_GC_PACKET_CHUNK_SIZE) {
-        LOGGER_ERROR(log, "Packet payload size (%u) exceeds maximum (%u)", length, MAX_GC_PACKET_CHUNK_SIZE);
-        return -1;
+    if (net_packet_type == NET_PACKET_GC_LOSSY) {
+        if (length > MAX_GC_CUSTOM_LOSSY_PACKET_SIZE) {
+            LOGGER_ERROR(log, "Packet payload size (%u) exceeds maximum (%u)", length, MAX_GC_CUSTOM_LOSSY_PACKET_SIZE);
+            return -1;
```

Comment:
Is this code unreachable because we check earlier? If not, can you write a test that covers this?